src/services/wbtcCafe.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import { Contract } from 'ethers';
+import { WBTC, BTC } from 'constants/assetsConstants';
+import { WBTC_PENDING_TRANSACTION } from 'constants/exchangeConstants';
+import { TX_PENDING_STATUS } from 'constants/historyConstants';
+import CURVE_ABI from 'abi/WBTCCurve.json';
+import { getEthereumProvider, reportLog } from 'utils/common';
+import type { WBTCFeesWithRate, WBTCFeesRaw, PendingWBTCTransaction } from 'models/WBTC';
+import type { Transaction } from 'models/Transaction';
+import { getEnv } from 'configs/envConfig';
+import Toast from 'components/Toast';
+import t from 'translations/translate';
+/* eslint-disable i18next/no-literal-string */
+
+// much of this function's body is copy-pasted from wbtc.cafe web app code
+export const gatherWBTCFeeData = async (
+  amount: number,
+  fees: ?WBTCFeesRaw,
+  fromAssetCode: string,
+): Promise<?WBTCFeesWithRate> => {
+  if (!fees || !fromAssetCode) return null;
+  const isSellingWbtc = fromAssetCode === WBTC;
+  const fixedFeeKey = isSellingWbtc ? 'release' : 'lock';
+  const dynamicFeeKey = isSellingWbtc ? 'burn' : 'mint';
+
+  const fixedFee = Number(fees[BTC.toLowerCase()][fixedFeeKey] / 100000000); // / 10 ** 8
+  const dynamicFeeRate = Number(
+    fees[BTC.toLowerCase()].ethereum[dynamicFeeKey] / 10000,
+  );
+
+  try {
+    const isProdEnv = getEnv().NETWORK_PROVIDER === 'homestead';
+    let exchangeRate;
+    let total;
+
+    const curve = new Contract(
+      getEnv().WBTC_CURVE,
+      CURVE_ABI,
+      getEthereumProvider(isProdEnv ? 'homestead' : 'kovan'),
+    );
+    const renVMFee = Number(amount) * dynamicFeeRate;
+    const amountAfterMint =
+        Number(amount - renVMFee - fixedFee) > 0
+          ? Number(amount - renVMFee - fixedFee)
+          : 0;
```

Comment:
This is the equal, but less noise:

```suggestion
    const amountAfterMint = Number(amount - renVMFee - fixedFee) || 0;
```